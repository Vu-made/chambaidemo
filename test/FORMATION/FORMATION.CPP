#include <bits/stdc++.h>
#define N 1000001
using namespace std;
int L[N], R[N];
int n, m;
void xuly1()
{
    int i,j;
    cin >> i >> j;
    L[i] = L[j];
    R[i] = j;
    R[L[j]] = i;
    L[j] = i;
}

void xuly2()
{
    int i,j;
    cin >> i >> j;
    L[i] = j;
    R[i] = R[j];
    L[R[j]] = i;
    R[j] = i;
}

void xuly3()
{
    int i;
    cin >> i;
    R[L[i]] = R[i];
    L[R[i]] = L[i];
}

void xuly4()
{
    int i;
    cin >> i;
    cout << L[i] << ' ' << R[i] << endl;
}
int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    freopen("FORMATION.INP","r",stdin);
    freopen("FORMATION.OUT","w",stdout);
    cin >> n >> m;
    for (int i=1; i<=m; i++)
    {
        int k;
        cin >> k;
        if (k == 1) xuly1();
        else if (k==2) xuly2();
        else if (k==3) xuly3();
        else xuly4();
    }
    return 0;
}
