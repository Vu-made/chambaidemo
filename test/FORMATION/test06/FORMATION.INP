51 99
4 1
4 1
2 5 1
1 50 1
2 32 50
1 45 32
3 50
4 5
1 13 45
4 32
2 38 1
1 51 32
4 13
4 45
2 21 5
4 32
2 9 21
2 6 13
2 28 21
1 31 32
2 49 32
3 13
1 19 45
2 3 1
4 49
4 1
1 24 45
2 42 32
3 45
2 30 3
1 2 42
3 24
1 11 3
4 49
2 15 32
3 1
3 5
2 12 30
4 19
1 33 30
3 51
1 22 21
3 19
3 30
4 31
4 31
4 9
2 44 3
4 21
3 21
1 46 44
3 12
1 40 6
4 42
1 4 2
1 43 38
4 32
3 44
3 31
1 8 9
2 29 42
3 32
2 34 15
4 34
4 42
1 25 2
1 17 15
4 15
2 10 8
2 23 43
1 30 33
4 8
2 27 17
2 45 23
1 39 42
1 16 29
1 24 22
1 35 16
3 11
4 29
3 35
1 1 23
3 25
3 39
1 37 45
1 5 30
4 1
1 31 27
1 19 2
3 16
2 48 49
2 18 5
4 33
2 51 15
1 20 51
3 15
3 28
2 50 8
2 44 37
