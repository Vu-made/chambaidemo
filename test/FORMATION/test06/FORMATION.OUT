0 0
0 0
1 0
45 1
0 45
13 51
51 1
32 1
49 3
42 1
6 51
6 32
6 32
28 0
22 28
2 49
31 15
15 4
2 29
17 34
28 10
16 49
43 23
30 43
